database/migrations/2025_12_29_000001_update_ticket_table.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    public function up(): void
+    {
+        // Drop the unique constraint that would block the data migration.
+        // This unique constraint isn't desirable - a user may open another ticket
+        // for the same achievement at some time in the future.
+        Schema::table('Ticket', function (Blueprint $table) {
+            $table->dropUnique('tickets_ticketable_reporter_id_index');
+        });
+
+        // Rename columns.
+        Schema::table('Ticket', function (Blueprint $table) {
+            $table->renameColumn('ID', 'id');
+            $table->renameColumn('ticketable_model', 'ticketable_type');
+            $table->renameColumn('ReportType', 'type');
+            $table->renameColumn('ReportState', 'state');
+            $table->renameColumn('ReportNotes', 'body');
+            $table->renameColumn('ReportedAt', 'created_at');
+            $table->renameColumn('ResolvedAt', 'resolved_at');
+            $table->renameColumn('Updated', 'updated_at');
+            $table->renameColumn('Hardcore', 'hardcore');
+        });
+
+        // Migrate AchievementID data to ticketable columns.
+        DB::statement(<<<SQL
+            UPDATE Ticket
+            SET ticketable_type = 'achievement',
+                ticketable_id = AchievementID
+            WHERE AchievementID IS NOT NULL
+              AND ticketable_id IS NULL
+        SQL);
+
+        // Drop indexes that reference AchievementID before dropping the column.
+        Schema::table('Ticket', function (Blueprint $table) {
+            $table->dropIndex('tickets_achievement_id_reporter_id_index');
+            $table->dropIndex('ticket_achievementid_reportstate_deleted_at_index');
+        });
+
+        // Drop the AchievementID column.
+        Schema::table('Ticket', function (Blueprint $table) {
+            $table->dropColumn('AchievementID');
+        });
+
+        // Rename the table.
+        Schema::rename('Ticket', 'trigger_tickets');
```

Comment:
I understand the reasoning behind this, but is it really necessary? It feels exceptionally verbose and possibly restricting. It would be like renaming `Comment` to `article_comments`.

Will we ever have tickets for things that aren't trigger-based? Things like "Unwelcome concept", "Incorrect type", or "Grammatical error" are ticket types, but would still apply to triggerable objects. It looks like the schema is already ready if we do: `ticketable_type` is not `trigger_type`.